### PR TITLE
Chemical - Fix chemical zeus module

### PR DESCRIPTION
**Pull request by AtrixZockt**
I hate this bs...

**When merged this pull request will:**
- _Describe what this pull request will do_
- _Each change in a separate line_

### IMPORTANT

- [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: addons/chemical/functions/fnc_gasCheck.sqf [hunk 1]
```diff
@@ -18,70 +17,89 @@
  * [logic, getpos player, 50, 20, 1] call kat_chemical_fnc_gasCheck;
  *
  * Public: No
-*/
+ */
+
+params ["_logic", "_pos", "_radius_max", "_radius_min", "_gastype"];
+
+systemChat "Gas check started";
```
Comment (YetheSamartaka) [inline addons/chemical/functions/fnc_gasCheck.sqf]: left systemChat from debugging

File: addons/chemical/functions/fnc_gasCheck.sqf [hunk 2]
```diff
@@ -18,70 +17,89 @@
...
 
-params ["_logic","_pos","_radius_max","_radius_min","_gastype"];
+systemChat str _logic;
```
Comment (YetheSamartaka) [inline addons/chemical/functions/fnc_gasCheck.sqf]: here as well

File: addons/chemical/functions/fnc_gasCheck.sqf [hunk 3]
```diff
@@ -18,70 +17,89 @@
...
-                private _arrPos = _playerARR find _x;
-                _playerARR deleteAt _arrPos;
-                _logic setVariable [QGVAR(gas_playerArr), _playerARR, true];
-            };
 
-            if (_x in _playerARR && _distance > _radius_max) then {
-                private _arrPos = _playerARR find _x;
-                _playerARR deleteAt _arrPos;
-                _logic setVariable [QGVAR(gas_playerArr), _playerARR, true];
+                systemChat format["Player treated: %1", _x];
```
Comment (YetheSamartaka) [inline addons/chemical/functions/fnc_gasCheck.sqf]: systemChat

File: addons/chemical/functions/fnc_gasCheck.sqf [hunk 4]
```diff
@@ -18,70 +17,89 @@
...
+                _logic setVariable [QGVAR(gas_playerArr), _playerArr, true];
             };
 
-            if (!(_x in _playerARR) && _distance < _radius_max) then {
-                _playerARR pushBack _x;
-                _logic setVariable [QGVAR(gas_playerArr), _playerARR, true];
+            if (!(_x in _playerArr) && _distance < _radius_max) then {
+                _playerArr pushBack _x;
+
+                systemChat format["Player in Zone: %1", _x];
```
Comment (YetheSamartaka) [inline addons/chemical/functions/fnc_gasCheck.sqf]: systemChat

File: addons/chemical/functions/fnc_gasCheck.sqf [hunk 5]
```diff
@@ -18,70 +17,89 @@
...
+
+                _logic setVariable [QGVAR(gas_playerArr), _playerArr, true];
                 [QGVAR(gasCheck_local), [_x, _logic, _pos, _radius_max, _radius_min, _gastype], _x] call CBA_fnc_targetEvent;
-            };
+            } else {
+                if (_x in _playerArr && _distance > _radius_max) then {
+
+                    systemChat format["Player left Zone: %1", _x];
```
Comment (YetheSamartaka) [inline addons/chemical/functions/fnc_gasCheck.sqf]: systemChat

File: addons/chemical/functions/fnc_gasCheckLocal.sqf [hunk 6]
```diff
@@ -23,103 +23,68 @@
 
 params ["_unit", "_logic", "_pos", "_radius_max", "_radius_min", "_gastype"];
 
+systemChat "Local Gas Check started";
```
Comment (YetheSamartaka) [inline addons/chemical/functions/fnc_gasCheckLocal.sqf]: systemChat

File: addons/chemical/functions/fnc_ui_gasModule.sqf [hunk 7]
```diff
@@ -10,24 +10,28 @@
...
  * Example:
- * [1105] call kat_chemical_fnc_spawnGasSmoke;
+ * [1105] call kat_chemical_fnc_ui_gasModule;
  *
  * Public: No
 */
 
 params ["_control"];
 
+systemChat "Initialize Module Test";
```
Comment (YetheSamartaka) [inline addons/chemical/functions/fnc_ui_gasModule.sqf]: systemChat

File: addons/chemical/functions/fnc_ui_gasModule.sqf [hunk 8]
```diff
@@ -10,24 +10,28 @@
...
+
 private _display = ctrlParent _control;
 private _ctrlButtonOK = _display displayCtrl IDC_OK;
 private _ctrlButtonCancel = _display displayCtrl IDC_CANCEL;
-private _logic = missionNamespace getVariable["BIS_fnc_initCuratorAttributes_target",objNull];
+private _logic = GETMVAR(BIS_fnc_initCuratorAttributes_target,objNull);
+
+systemChat str _ctrlButtonOK;
```
Comment (YetheSamartaka) [inline addons/chemical/functions/fnc_ui_gasModule.sqf]: systemChat

File: addons/chemical/functions/fnc_ui_gasModule.sqf [hunk 9]
```diff
@@ -71,10 +75,12 @@ if !(isNull attachedTo _logic) then {
 private _fnc_onConfirm = {
     params [["_ctrlButtonOK", controlNull, [controlNull]]];
 
+    systemChat "Module Confirm Click";
```
Comment (YetheSamartaka) [inline addons/chemical/functions/fnc_ui_gasModule.sqf]: systemChat

File: addons/chemical/functions/fnc_ui_gasModule.sqf [hunk 10]
```diff
@@ -93,29 +99,30 @@ private _fnc_onConfirm = {
...
-        private _logic = missionNamespace getVariable ["BIS_fnc_initCuratorAttributes_target",objNull];
+        private _logic = GETMVAR(BIS_fnc_initCuratorAttributes_target,objNull);
         if (isNull _logic) exitWith {};
 
         if !(isNull attachedTo _logic) then {
             private _object = attachedto _logic;
-            private _position = getPos _object;
 
-            [_logic,_position,_radius_max,_radius_min,_gastype] call FUNC(gasCheck);
+            systemChat "Module Gas Check Initialize";
```
Comment (YetheSamartaka) [inline addons/chemical/functions/fnc_ui_gasModule.sqf]: systemChat

File: addons/chemical/functions/fnc_ui_gasModule.sqf [hunk 11]
```diff
@@ -93,29 +99,30 @@ private _fnc_onConfirm = {
...
+            [_logic,getPos _object,_radius_max,_radius_min,_gastype] call FUNC(gasCheck);
 
             if (_display getVariable[QGVAR(ui_sealable),false]) then {
                 [_object] call FUNC(createSealAction);
             };
 
         } else {
-            private _position = getPos _logic;
-            [_logic,_position,_radius_max,_radius_min,_gastype] call FUNC(gasCheck);
+            systemChat "Module Gas Check Initialize";
```
Comment (YetheSamartaka) [inline addons/chemical/functions/fnc_ui_gasModule.sqf]: systemChat

**Review by YetheSamartaka — CHANGES_REQUESTED** [review]
Please remove systemChat messages.

**Review by YetheSamartaka — APPROVED** [review]

**Review by MiszczuZPolski — APPROVED** [review]

**Pull request merged**